src/transformers/modeling_utils.py

```diff
@@ -4120,15 +4120,38 @@ def _load_pretrained_model(
         expected_keys = list(model_state_dict.keys())
         prefix = model.base_model_prefix
 
+        error_msgs = []
+
+        old_keys = []
+        new_keys = []
+        renamed_keys = {}
+
+        warning_msg = f"This model {type(model)}"
+
+        original_loaded_keys = loaded_keys.copy()
+
         def _fix_key(key):
             if "beta" in key:
                 return key.replace("beta", "bias")
             if "gamma" in key:
                 return key.replace("gamma", "weight")
-            return key
-
-        original_loaded_keys = loaded_keys
-        loaded_keys = [_fix_key(key) for key in loaded_keys]
+            return None
+
+        for i, key in enumerate(loaded_keys):
+            new_key = _fix_key(key)
+            if new_key:
+                old_keys.append(key)
+                new_keys.append(new_key)
+                renamed_keys[key] = new_key
+                loaded_keys[i] = new_key
+
+        if renamed_keys:
+            warning_msg += 'contains parameters that have been renamed internally ("gamma" and "beta" in parameters) (a few are listed below but more are present in the model):\n'
+            logger.warning(warning_msg)
+            for old_key, new_key in renamed_keys.items():
+                warning_msg += f"* `{old_key}` -> `{new_key}`\n"
+            warning_msg += "If you are using a model from the Hub, consider submitting a PR to adjust these weights and help future users."
+            logger.info(warning_msg)
```

Comment:
This message isn't consistent - at the moment all of the renamed keys will be listed. Like in the other places where this logic is added, let's just take the first renames